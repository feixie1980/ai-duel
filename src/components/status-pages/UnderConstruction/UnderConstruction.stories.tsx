import type { Meta, StoryObj } from '@storybook/react';
import UnderConstruction from './UnderConstruction';

const meta = {
  title: 'Components/status-pages/UnderConstruction',
  component: UnderConstruction,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof UnderConstruction>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
