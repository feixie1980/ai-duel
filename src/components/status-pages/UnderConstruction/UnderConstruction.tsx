import React from 'react';
import styled from 'styled-components';
import { UnderConstructionProps } from './UnderConstruction.types';
import UnderConstructionImg from '../../../assets/under_construction_beach.png';
import { createGlobalStyle } from 'styled-components';
import DeliusUnicaseBold from '../../../assets/fonts/DeliusUnicase-Bold.ttf';

const FontStyle = createGlobalStyle`
  @font-face {
    font-family: 'DeliusUnicaseBold';
    src: url(${DeliusUnicaseBold}) format('truetype');
  }
`;

const StyledWrapper = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  overflow: hidden;
  font-family: 'DeliusUnicaseBold';
`;

const UCImage = styled.img`
  width: 50%;
`;

const UnderConstruction: React.FC = ({ className }: UnderConstructionProps) => {
  return (
    <>
      <FontStyle />
      <StyledWrapper className={className}>
        <h1>Under construction, coming soon!</h1>
        <UCImage src={UnderConstructionImg} />
      </StyledWrapper>
    </>
  );
};

export default UnderConstruction;
