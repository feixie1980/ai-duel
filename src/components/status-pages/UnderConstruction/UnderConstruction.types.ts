export interface UnderConstructionProps {
  /**
   * Custom class name for the container of the component.
   */
  className?: string;
}
