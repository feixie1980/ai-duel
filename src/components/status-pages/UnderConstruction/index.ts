export { default as UnderConstruction } from './UnderConstruction'; 
export type { UnderConstructionProps } from './UnderConstruction.types';