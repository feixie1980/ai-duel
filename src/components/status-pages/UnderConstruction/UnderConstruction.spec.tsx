import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import UnderConstruction from './UnderConstruction';

describe('UnderConstruction', () => {
  it('renders correctly', () => {
    const { container } = render(<UnderConstruction />);
    expect(container).toMatchSnapshot();
  });
});
