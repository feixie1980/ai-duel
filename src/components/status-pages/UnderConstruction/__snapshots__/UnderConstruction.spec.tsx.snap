// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnderConstruction > renders correctly 1`] = `
<div>
  <div
    class="UnderConstruction__StyledWrapper-sc-74c0038-0 dccqXF"
  >
    <img
      class="UnderConstruction__UCImage-sc-74c0038-1 iihBbY"
      src="/coding-ai-chat/src/assets/under_construction_beach.png"
    />
  </div>
</div>
`;
